### PR TITLE
Add support for additional builder arguments

### PR DESCRIPTION
**Pull request by regiswilson**
This addresses my comments in #15

### CODE REVIEW

**Comment by 72636c** [comment]
Heya, thanks for the PR! This looks sensible.

Thoughts on making this an array [like the Docker Compose plugin](https://github.com/buildkite-plugins/docker-compose-buildkite-plugin#build-arguments)?

**Comment by regiswilson** [comment]
> Thoughts on making this an array [like the Docker Compose plugin](https://github.com/buildkite-plugins/docker-compose-buildkite-plugin#build-arguments)?

I certainly wish that all the plugins would be similar enough (if not identical). The ecr-cache plugin uses the name "additional build args" as a plain string which I think is slightly better than when they use the "args" name in the docker plugin. In truth, args/build_args/additional_build_args needs to be cleared up before we press on. But that's a nit among gnats!

1. I have no problem making it available as an array optionally (but with the problem that I would then want to also patch the ecr-cache plugin).
2. It appears there is a merge conflict at the moment with a "target" parameter (similarly aped from the ecr-cache and other plugins). Funnily enough, I had a branch already setup for a similar PR I was going to send you after this one. I had targeted in this PR v1.3.0 which was released very recently, so I need to merge-conflict and re-target 1.4.0.

So I propose (if you accept): We move ahead with this PR (post merge-conflict) while also opening issues in both this repo and the ecr-cache repo to make them both accept arrays and strings [as well as perhaps identify what the best course of naming for these parameters would be --optional].

**Comment by regiswilson** [comment]
This PR has been updated to avoid a conflict with v1.3.0 and retargets v1.4.0

File: README.md [hunk 1]
```diff
@@ -89,6 +89,19 @@ steps:
           tags: prod-$BUILDKITE_BUILD_NUMBER
 ```
 
+Additional `docker build` arguments be passed via the `additional-build-args` setting:
+
+```
```
Comment (72636c) [inline README.md]:
```suggestion
Additional `docker build` arguments can be passed via the `additional-build-args` setting:

```yaml
```

File: README.md [hunk 2]
```diff
@@ -137,6 +150,10 @@ steps:
   instead of `MY_ARG=blah`), so that they are not checked into your source
   control and then logged to Buildkite output by this plugin.
 
+- `additional-build-args` (optional, string)
+
+  Allows specifying additional build arguments directly to the `docker` command.
```
Comment (72636c) [inline README.md]:
```suggestion
  Allows specifying additional arguments directly to the `docker build` command.
```

**Review by 72636c — APPROVED** [review]
I had another think about this one, let's just leave it as a string. Thanks again for the PR, I'll get this out after the `target` fix.

**Pull request merged**